src/server/plugins/initialize.js

```diff
@@ -40,6 +40,10 @@ module.exports = async function (kbnServer, server, config) {
     path.pop();
   };
 
+  for (let plugin of plugins) {
+    await plugin.preInit();
```

Comment:
Does pre-initialization not require to follow the same dependency graph as initialization does?